cpp/src/arrow/type_test.cc

```diff
@@ -364,152 +365,344 @@ TEST(TestField, TestMerge) {
+                                                       {out.v0.field, out.v1.field}));
+    ARROW_RETURN_NOT_OK(out.array->ValidateFull());
+    // Finalize the input RecordBatch
+    ARROW_ASSIGN_OR_RAISE(out.record_batch, RecordBatch::FromStructArray(out.array));
+    ARROW_RETURN_NOT_OK(out.record_batch->ValidateFull());
+    // Finalize the input ChunkedArray
+    out.chunked_array = SliceToChunkedArray(*out.array, out.num_chunks);
+    ARROW_RETURN_NOT_OK(out.chunked_array->ValidateFull());
+
+    // For each expected child array, create a chunked equivalent (we use a different
+    // chunk layout for each top-level column to make the Table test more interesting)
+    for (OutputValues* v :
+         {&out.v0, &out.v1, &out.v1_0, &out.v1_1, &out.v1_1_0, &out.v1_1_1,
+          &out.v1_0_flat, &out.v1_1_flat, &out.v1_1_0_flat, &out.v1_1_1_flat}) {
+      v->chunked_array =
+          SliceToChunkedArray(*v->array, out.num_column_chunks[v->path[0]]);
+    }
+    // Finalize the input Table
+    out.table =
+        Table::Make(out.schema, {out.v0.chunked_array, out.v1.chunked_array}, kNumRows);
+    ARROW_RETURN_NOT_OK(out.table->ValidateFull());
+
+    return std::move(out);
   }
-  EXPECT_RAISES_WITH_MESSAGE_THAT(Invalid, HasSubstr("empty indices cannot be traversed"),
-                                  FieldPath().Get(*table));
-}
-
-TEST(TestFieldPath, GetForChunkedArray) {
-  using testing::HasSubstr;
-
-  auto f0 = field("a", int32());
-  auto f1 = field("b", int32());
-  auto f2 = field("c", struct_({f1}));
-  auto f3 = field("d", struct_({f0, f2}));
-  auto type = struct_({f0, f1, f3});
-
-  auto column0 = ChunkedArrayFromJSON(f0->type(), {"[0,1,2,3]"});
-  auto column1 = ChunkedArrayFromJSON(f1->type(), {"[3,2,1,0]"});
-  auto column2_1 =
-      ChunkedArrayFromJSON(f2->type(), {R"([{"b":3},{"b":2},{"b":1},{"b":0}])"});
-  auto chunked_array = ChunkedArrayFromJSON(
-      type,
-      {
-          R"([{"a":0,"b":3,"d":{"a":0,"c":{"b":3}}}])",
-          R"([{"a":1,"b":2,"d":{"a":1,"c":{"b":2}}},{"a":2,"b":1,"d":{"a":2,"c":{"b":1}}}])",
-          R"([{"a":3,"b":0,"d":{"a":3,"c":{"b":0}}}])",
-      });
-  ASSERT_OK(chunked_array->ValidateFull());
-
-  ASSERT_OK_AND_ASSIGN(auto v0, FieldPath({0}).Get(*chunked_array));
-  ASSERT_OK_AND_ASSIGN(auto v1, FieldPath({1}).Get(*chunked_array));
-  ASSERT_OK_AND_ASSIGN(auto v2_0, FieldPath({2, 0}).Get(*chunked_array));
-  ASSERT_OK_AND_ASSIGN(auto v2_1, FieldPath({2, 1}).Get(*chunked_array));
-  ASSERT_OK_AND_ASSIGN(auto v2_1_0, FieldPath({2, 1, 0}).Get(*chunked_array));
-
-  for (const auto& v : {v0, v1, v2_0, v2_1, v2_1_0}) {
-    EXPECT_EQ(v->num_chunks(), chunked_array->num_chunks());
+
+  static std::shared_ptr<ChunkedArray> SliceToChunkedArray(const Array& array,
+                                                           int num_chunks) {
+    ARROW_CHECK(num_chunks > 0 && array.length() >= num_chunks);
```

Comment:
Generally we should favor `ASSERT` or `EXPECT` macros in tests, since they should arrow the tests to proceed (unless the test would otherwise crash just after the failed check).

Reply:
This function is only used to generate the shared test case so I figured that aborting here would be appropriate (otherwise, anything could happen).